src/routers/TrackingRouter.js

```diff
@@ -45,49 +45,49 @@ class TrackingRouter extends AppDevRouter<Object> {
                             },
                     };
 
-                    const trackingRequest = JSON.parse(await new Promise((resolve, reject) => {
+                    const trackingRequest = await new Promise((resolve, reject) => {
                         request(options, (error, response, body) => {
                             if (error) reject(error);
-                            console.log(response);
                             resolve(body);
                         });
                     }).then(value => value).catch((error) => {
                         ErrorUtils.log(error, null, 'Tracking request failed');
                         return null;
-                    }));
-
-                    console.log('tracking req', trackingRequest);
-
-                    const trackingData = trackingRequest.data.filter(busInfo => busInfo.VehicleId === realtimeData.vehicleID).map((busInfo) => {
-                        let lastUpdated = busInfo.LastUpdated;
-                        const firstParan = lastUpdated.indexOf('(') + 1;
-                        const secondParan = lastUpdated.indexOf('-');
-                        lastUpdated = parseInt(lastUpdated.slice(firstParan, secondParan));
-                        return {
-                            commStatus: busInfo.CommStatus,
-                            destination: busInfo.Destination,
-                            deviation: busInfo.Deviation,
-                            direction: busInfo.Direction,
-                            displayStatus: busInfo.DisplayStatus,
-                            gpsStatus: busInfo.GPSStatus,
-                            heading: busInfo.Heading,
-                            lastStop: busInfo.LastStop,
-                            lastUpdated,
-                            latitude: busInfo.Latitude,
-                            longitude: busInfo.Longitude,
-                            name: busInfo.Name,
-                            opStatus: busInfo.OpStatus,
-                            routeID: busInfo.RouteId,
-                            runID: busInfo.RunId,
-                            speed: busInfo.Speed,
-                            tripID: busInfo.TripId,
-                            vehicleID: busInfo.VehicleId,
-                            case: 'validData',
-                        };
                     });
 
+                    const trackingData = trackingRequest
+                        && JSON.parse(trackingRequest).data.filter(
+                            busInfo => busInfo.VehicleId === realtimeData.vehicleID,
+                        ).map((busInfo) => {
```

Comment:
Before you merge, can you make an issue to refactor this? It's pretty bulky & tough to understand